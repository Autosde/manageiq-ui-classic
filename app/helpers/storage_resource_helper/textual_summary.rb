module StorageResourceHelper::TextualSummary
  include TextualMixins::TextualGroupTags

  #
  # Groups
  #

  def textual_group_properties
    TextualGroup.new(
      _("Properties"),
      %i[
        name
        logical_free
        logical_total
        capabilities
      ]
    )
  end

  def textual_group_relationships
  end

  #
  # Items
  #

  def textual_name
    {:label => _("Name"), :value => @record.name}
  end

  def textual_logical_free
    {:label => _("Free Space"), :value => number_to_human_size(@record.logical_free, :precision => 2)}
  end

  def textual_logical_total
    {:label => _("Total Size"), :value => number_to_human_size(@record.logical_total, :precision => 2)}
  end

  def textual_capabilities
    values = ""
    JSON.parse(@record.capabilities).each do |capabilities|
      values += "#{capabilities['abstract_capability']}: #{capabilities['value']}, "
    end

    {:label => _("Enabled capabilities"), :value => values.chomp(', ') }
  end
end
