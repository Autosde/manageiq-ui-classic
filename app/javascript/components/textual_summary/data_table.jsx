/* eslint-disable react/no-array-index-key */
import * as React from 'react';
import PropTypes from 'prop-types';
import MiqDataTable from '../miq-data-table';

export default function DataTable(props) {
  const { rows, headers } = props;

  return (
    <MiqDataTable
      headers={headers}
      rows={rows}
      mode="data_table"
      onClick={() => props.onClick}
    />
  );
}

DataTable.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({ value: PropTypes.string.isRequired, sortable: PropTypes.string }),
  ])),
  rows: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.any)),
  onClick: PropTypes.func.isRequired,
};

DataTable.defaultProps = {
  rows: [],
  headers: '',
};
