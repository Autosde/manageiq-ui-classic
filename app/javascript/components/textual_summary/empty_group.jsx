import * as React from 'react';
import PropTypes from 'prop-types';

export default function EmptyGroup(props) {
  const { title, text } = props;
  return (
    <table className="table table-bordered table-striped table-summary-screen empty_group">
      <thead>
        <tr>
          <th>{title}</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>{text}</td>
        </tr>
      </tbody>
    </table>
  );
}

EmptyGroup.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};
