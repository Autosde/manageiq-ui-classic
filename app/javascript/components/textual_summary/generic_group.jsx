import * as React from 'react';
import PropTypes from 'prop-types';
import GenericTableRow from './generic_table_row';
import MiqStructuredList from '../miq-structured-list';

export default function GenericGroup(props) {
  const { title, items } = props;
  return (
    <>
      <MiqStructuredList
        rows={items}
        title={title}
        mode="generic_group"
        onClick={(item, event) => props.onClick(item, event)}
      />
      <table className="table table-bordered table-hover table-striped table-summary-screen generic_groups">
        <thead>
          <tr>
            <th colSpan="2" align="left">{title}</th>
          </tr>
        </thead>
        <tbody>
          {
            items.map((item, i) => (
            // eslint-disable-next-line react/no-array-index-key
              <GenericTableRow onClick={props.onClick} key={i} item={item} />
            ))
          }
        </tbody>
      </table>
    </>
  );
}

GenericGroup.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.any).isRequired,
  onClick: PropTypes.func.isRequired,
};
