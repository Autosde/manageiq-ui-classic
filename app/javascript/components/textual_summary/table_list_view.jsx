/* eslint-disable react/no-array-index-key */
import * as React from 'react';
import PropTypes from 'prop-types';
import MiqStructuredList from '../miq-structured-list';

export default function TableListView(props) {
  const { headers, values, title } = props;

  /** Function to generate rows for structured list. */
  const miqListRows = (list) => {
    const headerKeys = props.headers.map((item) => (item.key));
    return list.map((item) => (headerKeys.map((header) => item[header])));
  };

  return (
    <MiqStructuredList
      headers={headers.map((item) => item.label)}
      rows={miqListRows(values)}
      title={title}
      mode="table_list_view"
      onClick={() => props.onClick}
    />
  );
}

TableListView.propTypes = {
  title: PropTypes.string.isRequired,
  headers: PropTypes.arrayOf(PropTypes.any).isRequired,
  values: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClick: PropTypes.func.isRequired,
};
