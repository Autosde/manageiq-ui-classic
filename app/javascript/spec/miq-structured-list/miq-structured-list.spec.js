import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import MiqStructuredList from '../../components/miq-structured-list';
import { genericGroupData } from '../textual_summary/data/generic_group';
import { multilinkTableData } from '../textual_summary/data/multilink_table';
import { operationRangesData } from '../textual_summary/data/operation_ranges';
import { simpleTableData } from '../textual_summary/data/simple_table';
import { tableListViewData } from '../textual_summary/data/table_list_view';
import { tagGroupData } from '../textual_summary/data/tag_group';
import { summary1 } from '../textual_summary/data/textual_summary';

describe('Structured list component', () => {
  it('should render a simple list table', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<MiqStructuredList
      rows={genericGroupData.items}
      title={genericGroupData.title}
      mode="simple_table"
      onClick={onClick}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a multilink list table', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<MiqStructuredList
      rows={multilinkTableData.items}
      title={multilinkTableData.title}
      mode="multilink_table"
      onClick={onClick}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a operation range list table', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<MiqStructuredList
      rows={operationRangesData.items}
      title={operationRangesData.title}
      mode="operation_ranges"
      onClick={onClick}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a tag group list table', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<MiqStructuredList
      rows={tagGroupData.items}
      title={tagGroupData.title}
      mode="tag_group"
      onClick={onClick}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a simple table list table', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<MiqStructuredList
      rows={simpleTableData.items}
      title={simpleTableData.title}
      mode="simple_table"
      onClick={onClick}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
