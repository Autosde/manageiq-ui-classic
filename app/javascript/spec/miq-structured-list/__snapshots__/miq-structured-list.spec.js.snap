// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structured list component should render a multilink list table 1`] = `
<Fragment>
  <h3>
    Some OpenStack Stuff
  </h3>
  <FeatureToggle(StructuredListWrapper)
    ariaLabel="Structured list"
    className="miq-structured-list multilink_table"
  >
    <FeatureToggle(StructuredListBody)>
      <FeatureToggle(StructuredListRow)
        key="0"
        tabIndex={0}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Aodh
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=1&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Aodh"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Aodh"
                />
              </div>
              Running (3)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=1&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Aodh"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Aodh"
                />
              </div>
              All (3)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Aodh"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Aodh"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="1"
        tabIndex={1}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Ceilometer
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=2&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Ceilometer"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Ceilometer"
                />
              </div>
              Running (3)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=2&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Ceilometer"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Ceilometer"
                />
              </div>
              All (3)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Ceilometer"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Ceilometer"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="2"
        tabIndex={2}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Cinder
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=3&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Cinder"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Cinder"
                />
              </div>
              Running (3)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=3&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Cinder"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Cinder"
                />
              </div>
              All (3)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Cinder"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Cinder"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="3"
        tabIndex={3}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Glance
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=4&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Glance"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Glance"
                />
              </div>
              Running (2)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=4&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Glance"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Glance"
                />
              </div>
              All (2)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Glance"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Glance"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="4"
        tabIndex={4}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Gnocchi
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=5&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Gnocchi"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Gnocchi"
                />
              </div>
              Running (2)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=5&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Gnocchi"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Gnocchi"
                />
              </div>
              All (2)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Gnocchi"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Gnocchi"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="5"
        tabIndex={5}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Heat
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=6&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Heat"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Heat"
                />
              </div>
              Running (4)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=6&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Heat"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Heat"
                />
              </div>
              All (4)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Heat"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Heat"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="6"
        tabIndex={6}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Keystone
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Running (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=7&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Keystone"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Keystone"
                />
              </div>
              All (1)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Keystone"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Keystone"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="7"
        tabIndex={7}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Nova
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=8&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Nova"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Nova"
                />
              </div>
              Running (5)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=8&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Nova"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Nova"
                />
              </div>
              All (10)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Nova"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Nova"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="8"
        tabIndex={8}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          <div
            className="content"
          >
            Swift
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=9&status=running"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of running Swift"
              >
                <i
                  className="pficon pficon-ok"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of running Swift"
                />
              </div>
              Running (14)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            Failed (0)
          </div>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <a
            className="cell_link"
            href="/host/host_services/18?db=host&host_service_group=9&status=all"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="cell icon"
                title="Show list of all Swift"
              >
                <i
                  className="pficon pficon-service"
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                  title="Show list of all Swift"
                />
              </div>
              All (14)
            </div>
          </a>
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value sub_items"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
              title="Show list of configuration files of Swift"
            >
              <i
                className="fa fa-file-o"
                style={
                  Object {
                    "background": undefined,
                  }
                }
                title="Show list of configuration files of Swift"
              />
            </div>
            Configuration (0)
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
    </FeatureToggle(StructuredListBody)>
  </FeatureToggle(StructuredListWrapper)>
</Fragment>
`;

exports[`Structured list component should render a operation range list table 1`] = `
<Fragment>
  <h3>
    Operation Ranges
  </h3>
  <FeatureToggle(StructuredListWrapper)
    ariaLabel="Structured list"
    className="miq-structured-list operation_ranges"
  >
    <FeatureToggle(StructuredListBody)>
      <FeatureToggle(StructuredListRow)
        key="0"
        tabIndex={0}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          CPU
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            <div
              className="multi_row_cell"
            >
              <div
                className="sub_row_item"
                key="0"
              >
                <div
                  className="sub_label"
                >
                  Max
                </div>
                <div
                  className="sub_value"
                >
                  Not Available
                </div>
              </div>
              <div
                className="sub_row_item"
                key="1"
              >
                <div
                  className="sub_label"
                >
                  High
                </div>
                <div
                  className="sub_value"
                >
                  0 MHz
                </div>
              </div>
              <div
                className="sub_row_item"
                key="2"
              >
                <div
                  className="sub_label"
                >
                  Average
                </div>
                <div
                  className="sub_value"
                >
                  0 MHz
                </div>
              </div>
              <div
                className="sub_row_item"
                key="3"
              >
                <div
                  className="sub_label"
                >
                  Low
                </div>
                <div
                  className="sub_value"
                >
                  0 MHz
                </div>
              </div>
            </div>
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="1"
        tabIndex={1}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          CPU Usage
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            <div
              className="multi_row_cell"
            >
              <div
                className="sub_row_item"
                key="0"
              >
                <div
                  className="sub_label"
                >
                  Max
                </div>
                <div
                  className="sub_value"
                >
                  Not Available
                </div>
              </div>
              <div
                className="sub_row_item"
                key="1"
              >
                <div
                  className="sub_label"
                >
                  High
                </div>
                <div
                  className="sub_value"
                >
                  0.00%
                </div>
              </div>
              <div
                className="sub_row_item"
                key="2"
              >
                <div
                  className="sub_label"
                >
                  Average
                </div>
                <div
                  className="sub_value"
                >
                  0.00%
                </div>
              </div>
              <div
                className="sub_row_item"
                key="3"
              >
                <div
                  className="sub_label"
                >
                  Low
                </div>
                <div
                  className="sub_value"
                >
                  0.00%
                </div>
              </div>
            </div>
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
    </FeatureToggle(StructuredListBody)>
  </FeatureToggle(StructuredListWrapper)>
</Fragment>
`;

exports[`Structured list component should render a simple list table 1`] = `
<Fragment>
  <h3>
    Foo Bar
  </h3>
  <FeatureToggle(StructuredListWrapper)
    ariaLabel="Structured list"
    className="miq-structured-list simple_table"
  >
    <FeatureToggle(StructuredListBody)>
      <FeatureToggle(StructuredListRow)
        key="0"
        tabIndex={0}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          foo
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            bar
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="1"
        tabIndex={1}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          mrkef?
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            v zime!
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
    </FeatureToggle(StructuredListBody)>
  </FeatureToggle(StructuredListWrapper)>
</Fragment>
`;

exports[`Structured list component should render a simple table list table 1`] = `
<Fragment>
  <h3>
    Firewall Rules
  </h3>
  <FeatureToggle(StructuredListWrapper)
    ariaLabel="Structured list"
    className="miq-structured-list simple_table"
  >
    <FeatureToggle(StructuredListBody) />
  </FeatureToggle(StructuredListWrapper)>
</Fragment>
`;

exports[`Structured list component should render a tag group list table 1`] = `
<Fragment>
  <h3>
    Smart Management
  </h3>
  <FeatureToggle(StructuredListWrapper)
    ariaLabel="Structured list"
    className="miq-structured-list tag_group"
  >
    <FeatureToggle(StructuredListBody)>
      <FeatureToggle(StructuredListRow)
        key="0"
        tabIndex={0}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          Managed by Zone
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            <div
              className="cell icon"
            >
              <i
                className="pficon pficon-zone"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />
            </div>
            default
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="1"
        tabIndex={1}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          My Company X Tags
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            <div
              className="multi_row_cell"
            >
              <div
                className="sub_row_item"
                key="0"
              >
                <div
                  className="cell icon"
                >
                  <i
                    className="fa fa-tag"
                    style={
                      Object {
                        "background": undefined,
                      }
                    }
                  />
                </div>
                <div
                  className="sub_label"
                >
                  Dan Test
                </div>
                <div
                  className="sub_value"
                >
                  Test 1
                </div>
              </div>
              <div
                className="sub_row_item"
                key="1"
              >
                <div
                  className="cell icon"
                >
                  <i
                    className="fa fa-tag"
                    style={
                      Object {
                        "background": undefined,
                      }
                    }
                  />
                </div>
                <div
                  className="sub_label"
                >
                  Demo bla
                </div>
                <div
                  className="sub_value"
                >
                  Policy,2
                </div>
              </div>
            </div>
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
      <FeatureToggle(StructuredListRow)
        key="2"
        tabIndex={2}
      >
        <FeatureToggle(StructuredListCell)
          className="label_header"
        >
          Other Tags
        </FeatureToggle(StructuredListCell)>
        <FeatureToggle(StructuredListCell)
          className="content_value"
        >
          <div
            className="content"
          >
            <div
              className="multi_row_cell"
            >
              <div
                className="sub_row_item"
                key="0"
              >
                <div
                  className="cell icon"
                >
                  <i
                    className="fa fa-tag"
                    style={
                      Object {
                        "background": undefined,
                      }
                    }
                  />
                </div>
                <div
                  className="sub_label"
                >
                  Test
                </div>
                <div
                  className="sub_value"
                >
                  Test 1
                </div>
              </div>
              <div
                className="sub_row_item"
                key="1"
              >
                <div
                  className="cell icon"
                >
                  <i
                    className="fa fa-tag"
                    style={
                      Object {
                        "background": undefined,
                      }
                    }
                  />
                </div>
                <div
                  className="sub_label"
                >
                  Demo 2
                </div>
                <div
                  className="sub_value"
                >
                  Policy,2,3
                </div>
              </div>
              <div
                className="sub_row_item"
                key="2"
              >
                <div
                  className="sub_value"
                >
                  Demo 3
                </div>
              </div>
            </div>
          </div>
        </FeatureToggle(StructuredListCell)>
      </FeatureToggle(StructuredListRow)>
    </FeatureToggle(StructuredListBody)>
  </FeatureToggle(StructuredListWrapper)>
</Fragment>
`;
